xbout/load.py

```diff
@@ -91,33 +92,44 @@ def open_boutdataset(datapath='./BOUT.dmp.*.nc', inputfilepath=None,
     drop_variables : sequence, optional
         Drop variables in this list before merging. Allows user to ignore variables from
         a particular physics model that are not consistent between processors.
+    pre_squashed :  bool, optional
+        Set true when loading from data which was saved as separate variables
+        using ds.bout.save().
 
     Returns
     -------
     ds : xarray.Dataset
     """
 
-    # TODO handle possibility that we are loading a previously saved (and trimmed) dataset
-
-    # Determine if file is a grid file or data dump files
-    if _is_dump_files(datapath):
-        # Gather pointers to all numerical data from BOUT++ output files
-        ds = _auto_open_mfboutdataset(datapath=datapath, chunks=chunks,
-                                      keep_xboundaries=keep_xboundaries,
-                                      keep_yboundaries=keep_yboundaries,
-                                      drop_variables=drop_variables)
+    if pre_squashed:
+        ds = xr.open_mfdataset(datapath, chunks=chunks, combine='nested',
+                               concat_dim=None)
     else:
-        # Its a grid file
-        ds = _open_grid(datapath, chunks=chunks,
-                        keep_xboundaries=keep_xboundaries,
-                        keep_yboundaries=keep_yboundaries)
-
-    ds, metadata = _separate_metadata(ds)
-    # Store as ints because netCDF doesn't support bools, so we can't save bool
-    # attributes
-    metadata['keep_xboundaries'] = int(keep_xboundaries)
-    metadata['keep_yboundaries'] = int(keep_yboundaries)
-    ds = _set_attrs_on_all_vars(ds, 'metadata', metadata)
+        # Determine if file is a grid file or data dump files
+        if _is_dump_files(datapath):
+            # Gather pointers to all numerical data from BOUT++ output files
+            ds = _auto_open_mfboutdataset(datapath=datapath, chunks=chunks,
+                                          keep_xboundaries=keep_xboundaries,
+                                          keep_yboundaries=keep_yboundaries,
+                                          drop_variables=drop_variables)
+        else:
+            # Its a grid file
+            ds = _open_grid(datapath, chunks=chunks,
+                            keep_xboundaries=keep_xboundaries,
+                            keep_yboundaries=keep_yboundaries)
+
+        ds, metadata = _separate_metadata(ds)
+        # Store as ints because netCDF doesn't support bools, so we can't save
+        # bool attributes
+        metadata['keep_xboundaries'] = int(keep_xboundaries)
+        metadata['keep_yboundaries'] = int(keep_yboundaries)
+        ds = _set_attrs_on_all_vars(ds, 'metadata', metadata)
+
+        for var in _BOUT_TIME_DEPENDENT_META_VARS:
+            if var in ds:
+                # Assume different processors in x & y have same iteration etc.
+                latest_top_left = {dim: 0 for dim in ds[var].dims}
```

Comment:
Doesn't this select the earliest value (`t=0`, when `dim='t'`) of var rather than the latest?

Reply:
Yep, you're right